File: src/models/storage.model.js
const { Schema, model } = require('mongoose');

const storageSchema = new Schema({
    url_image: { type: String, required: true },
    name: { type: String, required: true },
    brand: { type: String, required: true },
    type: { type: String,  enum: ['SATA', 'NVMe'], required: true },
    capacity: { type: Number, required: true },
    speed: { type: Number, required: true },
    compatibility: {
        motherboard_storage_type: { type: String, enum: ['SATA', 'NVMe'], required: true }
    },
    url: { type: String, require: true },
    price: { type: Number, required: true}
});

module.exports = model('storage', storageSchema);

File: lib/models/storage.model.dart
class Storage {
  String name;
  String brand;
  String type;
  int capacity;
  int speed;
  Compatibility compatibility;
  String url;
  int price;

  Storage({
    required this.name,
    required this.brand,
    required this.type,
    required this.capacity,
    required this.speed,
    required this.compatibility,
    required this.url,
    required this.price,
  });
  factory Storage.fromJson(Map<String, dynamic> json) {
    return Storage(
      name: json['name'],
      brand: json['brand'],
      type: json['type'],
      capacity: json['capacity'],
      speed: json['speed'],
      compatibility: Compatibility.fromJson(json['compatibility']),
      url: json['url'],
      price: json['price'],
    );
  }
}

class Compatibility {
  String motherboardStorageType;

  Compatibility({
    required this.motherboardStorageType,
  });

  factory Compatibility.fromJson(Map<String, dynamic> json) {
    return Compatibility(
      motherboardStorageType: json['motherboard_storage_type'],
    );
  }
}
